src/components/BaseMarkdown.vue

```diff
@@ -18,9 +19,17 @@ const remarkable = new Remarkable({
 
 const markdown = computed(() => {
   let body = props.body;
-  body = remarkable.render(body);
-  // body = sanitizeHtml(body);
-  return body;
+
+  // Add the ipfs gateway to images that start with ipfs://
+  let splitBodyText = body.split(/(?=\))|(?<=\()/);
```

Comment:
This splits the text at any `(` or `)` if I'm not mistaken. This could be a bit more specific. Also the replace function excepts a function as the second argument, allowing for a call to getIpfsUrl to perform the replacement.

More specific regex, looking for the full markdown image pattern and capturing the link: `!\[.*?\]\((ipfs:\/\/[a-zA-Z0-9]+?)\)`

In combination with the [replace function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_a_parameter) this can be a one-liner.